modules/puppet_proxy/puppet_api.rb

```diff
@@ -45,10 +50,19 @@ class Proxy::Puppet::Api < ::Sinatra::Base
       class_retriever.classes_in_environment(params[:environment]).map{|k| {k.to_s => { :name => k.name, :module => k.module, :params => k.params} } }.to_json
     rescue Proxy::Puppet::EnvironmentNotFound
       log_halt 404, "Could not find environment '#{params[:environment]}'"
+    end
```

Comment:
You're no longer catching the ```Proxy::Puppet::TimeoutError``` above. Is that intentional?

Reply:
It is. It's not possible to timeout on this call, as the counters are updated asynchronously.